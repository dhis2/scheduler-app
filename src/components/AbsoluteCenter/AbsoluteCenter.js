import React from 'react'
import { node, bool } from 'prop-types'
import styles from './AbsoluteCenter.module.css'

/**
 * Centers its children in the viewport vertically and horizontally. Children are not clickable due
 * to pointer-events: none, to allow underlying elements to still respond to clicks. Children are
 * rendered in a row by default.
 */

const AbsoluteCenter = ({ children, vertical }) => {
    const classNames = [styles.center]

    if (vertical) {
        classNames.push(styles.vertical)
    }

    return <div className={classNames.join(' ')}>{children}</div>
}

AbsoluteCenter.defaultProps = {
    vertical: false,
}

AbsoluteCenter.propTypes = {
    children: node.isRequired,
    vertical: bool,
}

export default AbsoluteCenter
