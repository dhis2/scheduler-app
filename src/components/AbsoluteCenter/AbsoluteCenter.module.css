.center {
    display: flex;
    align-items: center;
    justify-content: center;
    position: fixed;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    pointer-events: none;
}

.clickable {
    pointer-events: auto;
    display: flex;
    align-items: center;
    justify-content: center;
}

.vertical {
    flex-direction: column;
}
