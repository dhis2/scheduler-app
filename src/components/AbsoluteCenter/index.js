import AbsoluteCenter from './AbsoluteCenter'

export { AbsoluteCenter }
