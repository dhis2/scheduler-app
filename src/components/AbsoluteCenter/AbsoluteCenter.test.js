import React from 'react'
import { shallow } from 'enzyme'
import AbsoluteCenter from './AbsoluteCenter'

describe('<AbsoluteCenter>', () => {
    it('renders correctly', () => {
        const wrapper = shallow(<AbsoluteCenter>Text</AbsoluteCenter>)

        expect(wrapper).toMatchSnapshot()
    })

    it('responds to a vertical prop', () => {
        const wrapper = shallow(<AbsoluteCenter vertical>Text</AbsoluteCenter>)

        expect(wrapper).toMatchSnapshot()
    })
})
