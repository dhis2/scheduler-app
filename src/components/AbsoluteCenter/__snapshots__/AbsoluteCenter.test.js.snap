// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AbsoluteCenter> renders correctly 1`] = `
<div
  className="center"
>
  Text
</div>
`;

exports[`<AbsoluteCenter> responds to a vertical prop 1`] = `
<div
  className="center vertical"
>
  Text
</div>
`;
